go/mysql/conn.go

```diff
@@ -718,6 +720,56 @@ func (c *Conn) writeErrorPacketFromError(err error) error {
 	return c.writeErrorPacket(ERUnknownError, SSUnknownSQLState, "unknown error: %v", err)
 }
 
+func (c *Conn) writeLoadInfilePacket(fileName string) error {
+	length :=  1 + len(fileName)
+	data := c.startEphemeralPacket(length)
+	pos := 0
+	pos = writeByte(data, pos, LocalInfilePacket)
+	pos = writeEOFString(data, pos, fileName)
+
+	return c.writeEphemeralPacket()
+}
+
+func (c *Conn) HandleLoadDataLocalQuery(tmpdir string, file string) error {
+	// First send the load infile packet and flush the connector
+	err := c.writeLoadInfilePacket(file)
+	if err != nil {
+		return err
+	}
+
+	err = c.flush()
+	if err != nil {
+		return err
+	}
+
+	// Wait for the response packet that contains the data
+	fileData, err := c.readEphemeralPacket()
+	if err != nil {
+		return err
+	}
+
+	// TODO: Change this to something that makes sense
+	fileName := filepath.Join(tmpdir, "random")
+
+	// Write the file to a temporary directory.
+	// TODO: Get the tmpdir config params from here
+	err = ioutil.WriteFile(fileName, fileData, 0644)
```

Comment:
There can be more than one packet with file data, right?

Reply:
Good point. Is the idea here to keep waiting with a timeout or something like that?